src/controller/http-api-router.js

```diff
@@ -23,44 +23,44 @@ class HttpApiRouter {
 
     async initializeListeners() {
         // POST REQUESTS
-        this.httpClientModule.post(HTTP_API_ROUTES.PUBLISH, (req, res) => {
+        this.httpClientModuleInterface.post(HTTP_API_ROUTES.PUBLISH, (req, res) => {
             this.publishController.handleHttpApiPublishRequest(req, res);
         });
 
-        this.httpClientModule.post(HTTP_API_ROUTES.PROVISION, (req, res) => {
+        this.httpClientModuleInterface.post(HTTP_API_ROUTES.PROVISION, (req, res) => {
             this.publishController.handleHttpApiProvisionRequest(req, res);
         });
```

Comment:
Please consider removing unnecessary wrapper function for controller handler.

Instead of 
```
this.httpClientModuleInterface.post(HTTP_API_ROUTES.PROVISION, (req, res) => {
            this.publishController.handleHttpApiProvisionRequest(req, res);
});
```
we can just have
```
this.httpClientModuleInterface.post(HTTP_API_ROUTES.PROVISION, this.publishController.handleHttpApiProvisionRequest);
```